cmd/ttn-lw-cli/commands/end_device_templates.go

```diff
@@ -403,7 +403,7 @@ command to assign EUIs to map to end device templates.`,
 				for _, e := range mapping {
 					switch {
 					case e.MappingKey != "" && e.MappingKey == inputEntry.MappingKey:
-					case e.EndDevice.ApplicationId != "" && e.EndDevice.ApplicationId == inputEntry.EndDevice.ApplicationId &&
+					case e.EndDevice.ApplicationIds.ApplicationId != "" && e.EndDevice.ApplicationIds.ApplicationId == inputEntry.EndDevice.ApplicationIds.ApplicationId &&
```

Comment:
I'm not entirely sure if the `ApplicationIds` will not be nil here since I don't see any validation in this file. Does the CLI validate the template that it decodes?

Reply:
Why not use the getters? With `e.GetEndDevice().GetApplicationIds().GetApplicationId()` you'll just get an empty string if anything in between is `nil`, and we already have a check for that.

Reply:
Nice. I didn't know we even had this.